pkg/client/defaults.go

```diff
@@ -42,7 +44,6 @@ func NewRunConfig() *RunConfig {
 // NewDeleteConfig is a DeleteConfig using default images, RBAC enabled, and DeleteAll enabled.
 func NewDeleteConfig() *DeleteConfig {
 	return &DeleteConfig{
-		Namespace:  config.DefaultImage,
```

Comment:
We should fix this, not remove the namespace defaulting. Clearly if no one has complained they aren't using this but if we are going to leave the `NewXyzConfig()` methods then it seems like we should (1) be consistent about setting namespace (2) seems like using `NewXyzConfig()` should pass the new validation routine we're making.

w.r.t. (1) the config.New() method, NewGenConfig, and NewRunConfig would all have their namespaces already set so I think we should do the same here.

I think we talked about this but refresh my memory if we came to a different conclusion; I know we didnt wan't to push CLI stuff here, but it does seem like we can provide the sane default so that it passes validation.

Reply:
We did talk about it. I think we were leaning towards eventually removing these if they weren't being used. In the meantime though, they should pass the validation, and given that the `config.New()` results in the `DefaultNamespace` being used, it makes sense for it to be used here for these configs. I'll fix it!